### DISCUSSION TITLE
Some trash in the game folders

### DISCUSSION THREAD

**Discussion by Dok11** [mcarlucci/decky-storage-cleaner#8] [Ideas]
Hi @mcarlucci ! Thanks for this great plugin!

Intresting what do you think about next case. In the "The Witcher 3" installed game of my Steam Deck I found some unnecessary files which took around 5 GB (this is pretty too much even in 512 GB version, imho) — soundtrack, wallpappers, art-books, pdf guides, comics in pdf, all language versions, etc.
So, how do you think, is cleaning such files (in games in common, not only the Witcher) might be a part of this plugin? Is this task in it's scope?
If no, you can stop reading here =)

I think it proboably might be like a json like:
```json
{
  "games-clean-config": [
    {
      "title": "The Witcher 3: Wild Hunt",
      "path": [
        "/home/.steam/steam/steamapps/common/The Witcher 3/"
      ],
      "trash-groups": [
        {
          "title": "Art-book",
          "path": [
            "./ARTBOOK/"
          ]
        },
        {
          "title": "Artwork",
          "path": [
            "./ARTWORK/"
          ]
        },
        {
          "title": "Manual",
          "path": [
            "./MANUAL/"
          ]
        },
        {
          "title": "Epilepsy Warning",
          "path": [
            "./content/content0/movies/cutscenes/gamestart/epilepsy/epilepsy_%USER_LANG%.usm"
          ]
        },
        {
          "title": "Localizations",
          "path": [
            "./content/content3/%USER_LANG%.w3strings"
          ]
        }
      ],
      "mask-regexp": [
        {
          "code": "%USER_LANG%",
          "value": "\\w+"
        }
      ],
      "path-masks": [
        {
          "code": "%USER_LANG%",
          "title": "Which language version we can remove"
        }
      ]
    }
  ]
}
```

With this config in the plugin options menu we might show founded games that represented in the config. In the game section show list of folders and files that might be removed and checkboxes for each of them.
Last section of "trash-groups" in my example contains a code in the path `%USER_LANG%` and "path-masks" in the next part. Here we could show the "title" as a question with a list of all founded values for this code. For example, in this case we can show:
- [ ] ar
- [ ] br
- [ ] cn
- [ ] cz
- [ ] de
- [ ] en
- [ ] es
- [ ] esmx
- [ ] fr
- [ ] hu
- [ ] it
- [ ] jp
- [ ] kr
- [ ] pl
- [ ] ru
- [ ] tr
- [ ] zh

Here user can choose which versions can be removed.

**Comment by Dok11** [comment]
Actually there is around 11 GB what I safely clear from just a single Witcher 3.
And here a related bug about displaying game size https://steamcommunity.com/app/1675200/discussions/1/6490302633982132777/

**Comment by mcarlucci** [comment]
@Dok11 Thanks for digging around and finding theses files/folders! I think the main issue that I have seen during my investigations is the inconsistencies of where and how game devs are storing additional assets. So it's hard to create a reliable and generic method to do this for all games in a consistent way. 

I think it may still be worth trying to create a predefined mapping/lookup table to look for file/folders names in the game directory and then attempting to clean them up that way.
